src/core/csv/reader.cc

```diff
@@ -789,8 +789,8 @@ void GenericReader::detect_and_skip_bom() {
 void GenericReader::skip_initial_whitespace() {
   const char* ch = sof;
   if (!sof) return;
-  while ((ch < eof) && (*ch <= ' ') &&
-         (*ch==' ' || *ch=='\n' || *ch=='\r' || *ch=='\t')) {
+  while ((ch < eof) && (*ch <= ' ')  &&
+         (*ch==' ' || *ch=='\n' || *ch=='\r' || (*ch=='\t' && *(ch+1) <= ' '))) {
```

Comment:
Need to also check that `(ch+1) < eof`, otherwise, you might be dereferencing an invalid pointer.

Reply:
Done